Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -141,6 +141,28 @@ namespace AzToolsFramework::Prefab
             FocusOnPrefabInstanceOwningEntityId(entityId);
         }
 
+        // Set the selection to the entities owned by the newly focused prefab instance.
+        // Only do it if the focused prefab is not the root.
+        if (auto focusedInstance = GetInstanceReference(m_rootAliasFocusPath);
+            focusedInstance.has_value() && focusedInstance->get().GetParentInstance() != AZStd::nullopt)
+        {
+            EntityIdList selectedEntities;
```

Comment:
suggestion: Would it be possible to request the number of NestedEntityIds first and then reserve this vector?

Reply:
I agree but I don't have a way to get the number of nested entities/instances without looping over them, so it would likely be counterproductive with the current API.